encode_test.go

```diff
@@ -45,14 +45,10 @@ func javaDecodeValidate(method string, target interface{}) (string, error) {
 	cmd := exec.Command("java", "-jar", hessianJar, method)
 
 	stdin, _ := cmd.StdinPipe()
-	_, e = stdin.Write(b)
-	if e != nil {
-		return "", e
-	}
-	e = stdin.Close()
-	if e != nil {
-		return "", e
-	}
+	go func() {
+		stdin.Write(b)
```

Comment:
why run in new goroutine?

Reply:
Firstly `StdinPipe` has byte number limit. So if a process fills the pipe fully by writing bytes and no process read bytes from the pipe, then the writing process will hang. One can find explicit limit number [here](https://stackoverflow.com/a/50118778/5772365).

Origin code is below:

https://github.com/dubbogo/hessian2/blob/a7ee7ab1292618d576895782e91fb8491ef50e49/encode_test.go#L45-L57

We can see that the writing process is the go program and the reading process is the java program. Further more go program starts writing from `line 48` and java program starts reading from `line 57`.

So if byte's number is larger than 4096 on windows, i.e. like [here](https://github.com/dubbogo/hessian2/pull/68/files#diff-31acfea64559ab28bbb17d31d8e00de0R142), go program will hang on write, and the java program will never start reading.

So we should run a writing goroutine to avoid hanging the reading process.